internal/server/authentication.go

```diff
@@ -0,0 +1,147 @@
+package server
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"strings"
+	"time"
+
+	"golang.org/x/crypto/bcrypt"
+)
+
+type joseHeader struct {
+	Alg string `json:"alg"`
+	Typ string `json:"typ"`
+}
+
+type jwsPayload struct {
+	UserID int `json:"user_id"`
+	Exp    int `json:"exp"`
+}
+
+// JWT is a struct that holds the relevant data for handling JWTs.
+type JWT struct {
+	Header    joseHeader
+	Payload   jwsPayload
+	Signature []byte
+	Raw       string
+}
+
+// GenerateJWT takes a user id and the signing secret and generates a JWT.
+// with the following structure:
+// Header: {"alg":"HS256", "typ":"JWT"}
+// Claims: {"user_id": user_id, "exp": [current time + 12hrs]}
```

Comment:
```suggestion
//  Header: {"alg":"HS256", "typ":"JWT"}
//  Claims: {"user_id": user_id, "exp": [current time + 12hrs]}
```

This will render these lines as a code block in the go docs and the linter will stop yelling at you https://tip.golang.org/doc/comment#code